### DISCUSSION TITLE
Metadata Updated At?

### DISCUSSION THREAD

**Discussion by eecue** [RhetTbull/osxphotos#910] [Q&A] [answered]
Is there a way to tell when metadata has been updated? I see there is a `date_modified` field but that stays `None` when changing things like Location or Keywords.

**Comment by RhetTbull** [comment, marked as answer]
Not sure as I've not looked into this specifically. The modification date is only updated when the photo is edited. 

You can do a test by running `osxphotos snap`, add a keyword, then run `osxphotos diff`. This will show all changes in the library. The `ACHANGE` values appear to be some sort of transaction log or rollback record but they won't be reliable for detecting this as I believe they're transient. You want to look for a change in `ZASSET`, `ZADDITIONALASSETATTRIBUTES` etc

If there is a record of the change date osxphotos can easily be modified to read the appropriate column.

**Reply by RhetTbull** [reply]
By the way, `osxphotos export` detects metadata changes (needed for `--update` ) by storing a JSON representation of all the metadata in the export database and comparing the current metadata with the stored metadata. This works for detecting changes but does not tell you *when* something changed, only that it changed.

**Reply by eecue** [reply]
that's actually perfect for what I'm doing... I don't really need to know when

**Reply by RhetTbull** [reply]
See also [hexdigest](https://github.com/RhetTbull/osxphotos/blob/master/API_README.md#hexdigest)

**Reply by eecue** [reply]
Oh yeah that's what I need!

**Reply by RhetTbull** [reply]
If you need an easy way to store the metadata or hex digest between runs, osxphotos includes a super simple Key-Value store backed by sqlite:  `from osxphotos.sqlitekvstore import SQLiteKVStore`.  For an example, see [detect_qrcodes.py](https://github.com/RhetTbull/osxphotos/blob/master/examples/detect_qrcodes.py)  Full docs [here](https://github.com/RhetTbull/sqlitekvstore)
